### PR TITLE
Refactor ANC module to use Hilt DI

### PR DESCRIPTION
**Pull request by ekigamba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Release | Other) 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/anc/src/main/java/org/smartregister/fhircore/anc/data/family/FamilyDetailRepository.kt [hunk 1]
```diff
@@ -32,11 +32,13 @@ import org.smartregister.fhircore.engine.util.DispatcherProvider
...
-  private val fhirEngine: FhirEngine,
-  private val dispatcherProvider: DispatcherProvider
+  val fhirEngine: FhirEngine,
+  val familyItemMapper: FamilyItemMapper,
+  val dispatcherProvider: DispatcherProvider
 ) {
 
+  lateinit var familyId: String
+
   fun fetchDemographics(): LiveData<Patient> {
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/data/family/FamilyDetailRepository.kt:42]: Can we remove the `LiveData` return type from the Repository

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/encounters/EncounterListActivity.kt [hunk 2]
```diff
@@ -18,25 +18,26 @@ package org.smartregister.fhircore.anc.ui.anccare.encounters
...
     super.onCreate(savedInstanceState)
 
     val patientId =
       intent.extras?.getString(QuestionnaireActivity.QUESTIONNAIRE_ARG_PATIENT_KEY) ?: ""
-    val repository = EncounterRepository((application as AncApplication).fhirEngine, patientId)
-    val viewModel = EncounterListViewModel.get(this, application as AncApplication, repository)
-
-    viewModel.setAppBackClickListener(this::handleBackClicked)
+    encounterListViewModel.repository.patientId = patientId
+    encounterListViewModel.setAppBackClickListener(this::handleBackClicked)
```
Comment (ellykits) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/encounters/EncounterListActivity.kt:38]: Can we refactor this so we can remove activity references on the viewmodel?

File: android/anc/src/main/java/org/smartregister/fhircore/anc/data/family/FamilyDetailRepository.kt (hunk 1 above)
Comment (ekigamba) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/data/family/FamilyDetailRepository.kt:42]: Yea man. Let me do that

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/encounters/EncounterListActivity.kt (hunk 2 above)
Comment (ekigamba) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/encounters/EncounterListActivity.kt:38]: Cool :+1:

**Pull request merged**